test/AshToken.js

```diff
@@ -0,0 +1,174 @@
+const chai = require("chai");
+const chaiAsPromised = require("chai-as-promised");
+const { expect } = chai;
+chai.use(chaiAsPromised);
+const BN = require("bn.js");
+chai.use(require("chai-bn")(BN));
+const { expectEvent } = require("openzeppelin-test-helpers");
+
+const ashToken = artifacts.require("./AshToken.sol");
+
+const { makeRandomAddress } = require("./utils");
+
+contract("AshToken", accounts => {
+  let testContract;
+
+  before(async () => {
+    testContract = await ashToken.deployed();
+  });
+
+  describe("decimals()", () => {
+    it("has 12 decimals", async () => {
+      const result = await testContract.decimals();
+      expect(result).to.be.a.bignumber.that.equals("12");
+    });
+  });
+
+  describe("symbol()", () => {
+    it("has symbol ASH", async () => {
+      const result = await testContract.symbol();
+      expect(result).to.eql("ASH");
+    });
+  });
+
+  describe("name()", () => {
+    it("has name Ash Token", async () => {
+      const result = await testContract.name();
+      expect(result).to.eql("Ash Token");
+    });
+  });
+
+  describe("totalSupply()", () => {
+    it("has initial supply of 0", async () => {
+      const result = await testContract.totalSupply();
+      expect(result).to.be.a.bignumber.that.equals("0");
+    });
+  });
+
+  describe("balanceOf()", () => {
+    it("has initial balance of 0", async () => {
+      const address = makeRandomAddress();
+      const balance = await testContract.balanceOf(address);
+      expect(balance).to.be.a.bignumber.that.equals("0");
+    });
+  });
+
+  describe("mint()", () => {
```

Comment:
Good test, can you also grab another account and ensure it doesn't work with the wrong minter?

`expect(testContract.mint(address, 7, {from: address})).to.be.rejected()` or something like that

Reply:
https://github.com/iov-one/ethereum-smart-contracts/pull/8/commits/6843b6d3d7dcb7bcd2b45a6af75b5bf51ca74c97